lib/msal-common/src/request/AuthenticationHeaderParser.ts

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { ClientConfigurationError } from "..";
+import { HeaderNames } from "../utils/Constants";
+
+type WWWAuthenticateChallenges = {
+    nonce?: string,
+};
+
+type AuthenticationInfoChallenges = {
+    nextnonce?: string
+};
+
+/**
+ * This is a helper class that parses supported HTTP response authentication headers to extract and return
+ * header challenge values that can be used outside the basic authorization flows.
+ */
+export class AuthenticationHeaderParser{
+    private headers: Headers;
```

Comment:
Is `Headers` here this? https://developer.mozilla.org/en-US/docs/Web/API/Headers

Reply:
I would say use a different type that is browser/node agnostic, given this type doesn't necessarily exist in a Node app.

Reply:
It's the one from `isomorphic-fetch`, which is the only way I found to "mock" the Headers class in this environment.

Reply:
Such as this: https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/51669adf5e2fe446bf94925fe77762b64e9513fc/lib/msal-common/src/network/INetworkModule.ts#L12-L15

Reply:
Nevermind, here it's the one from Fetch API. Thought this was the test file. Ok I'll try to make it platform agnostic.

Reply:
Also keep in mind an app may not use `fetch` for their network requests, so a generic string/string dictionary should be good enough here.

Reply:
Moved to use `NetworkRequestOptions` to keep it generic